### PR TITLE
added missing options

### LINKED ISSUE #3388: Include restore options in documentation for command with implicit restore
_From @evil-shrike on October 12, 2017 22:5_

I run dotnet publish to create a msdeploy package. I supplying an `-s` option with my NuGet private server. But I made a mistake in the repository url and got 404 error. It expected. But it means that nuget tried to restore my package from my nuget server.
```
C:\Work>dotnet publish -c Debug -s "https://company/api/nuget/rnd-virtual-internal" /p:WebPublishMethod=Package /p:DeployIisAppPath=mysite /p:PublishProfile=profile
C:\Program Files\dotnet\sdk\2.0.2\NuGet.targets(102,5): error : Failed to retrieve information about 'MyPackage' from remote source 'https://company/api/nuget/rnd-virtual-internal/FindPackagesById()?id='MyPackage''.\r [C:\Work\Server.csproj]
C:\Program Files\dotnet\sdk\2.0.2\NuGet.targets(102,5): error :   Response status code does not indicate success: 404 (Not Found). [C:\Work\Server.csproj]
```

Then I fixed error in the url and repeated:
```
C:\Work>dotnet publish -c Debug -s "https://company/api/nuget/nuget-virtual-internal" /p:WebPublishMethod=Package /p:DeployIisAppPath=mysite /p:PublishProfile=profile

Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1001: Unknown switch.
Switch: -s
```
but what I got is MSBUILD error for my `-s` option. 
I would understand if dotnet publish doesn't support `--source` option at all but if so why 404 happened with incorrect url?

Previously on netfx I used `/p:PackageSources="source"` property but now on `dotnet publish` it's ignored.

## Environment data
`dotnet --info` output:
D:\Work>dotnet --info
.NET Command Line Tools (2.0.2)

Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.2\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

_Copied from original issue: dotnet/cli#7825_

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #3388 

Also fixed the alphabetical order for options in the dotnet publish topic.

/cc @evil-shrike @livarcocc 

Internal review URLs:
[dotnet publish](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?branch=pr-en-us-3390&tabs=netcore2x)
[dotnet build](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build?branch=pr-en-us-3390&tabs=netcore2x)
[dotnet pack](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-pack?branch=pr-en-us-3390&tabs=netcore2x)
[dotnet run](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-run?branch=pr-en-us-3390&tabs=netcore2x)

### CODE REVIEW

File: docs/core/tools/dotnet-publish.md [hunk 1]
```diff
@@ -88,13 +88,17 @@ Doesn't perform an implicit restore when running the command.
 Specifies the path for the output directory. If not specified, it defaults to *./bin/[configuration]/[framework]/* for a framework-dependent deployment or *./bin/[configuration]/[framework]/[runtime]* for a self-contained deployment.
 If a relative path is provided, the output directory generated is relative to the project file location, not to the current working directory.
 
+`-r|--runtime <RUNTIME_IDENTIFIER>`
+
+Publishes the application for a given runtime. This is used when creating a [self-contained deployment (SCD)](../deploying/index.md#self-contained-deployments-scd). For a list of Runtime Identifiers (RIDs), see the [RID catalog](../rid-catalog.md). Default is to publish a [framework-dependent deployment (FDD)](../deploying/index.md#framework-dependent-deployments-fdd).
```
Comment (rpetrusha) [inline docs/core/tools/dotnet-publish.md]: nit: Default --> The default

**Review by rpetrusha — APPROVED** [review]
LGTM, @mairaw. Just one nit.

File: docs/core/tools/dotnet-publish.md (hunk 1 above)
Comment (mairaw) [reply docs/core/tools/dotnet-publish.md]: Thanks @rpetrusha. I actually added "The default behavior"

**Comment by mairaw** [comment]
@livarcocc looks good? ok to merge?

**Comment by livarcocc** [comment]
I am not sure that's enough. These commands support all options from restore, not only source.

cc @KathleenDollard

**Comment by mairaw** [comment]
Gotcha @livarcocc. It wasn't clear to me there were other options missing from the issue description. See if it looks good now.

/cc @KathleenDollard

**Comment by livarcocc** [comment]
Should we do this for dotnet run, dotnet pack and dotnet test as well. As these two also support implicit restore.

**Comment by mairaw** [comment]
Sure. I'll add to those too.

**Comment by mairaw** [comment]
@livarcocc sorry for the delay. I've added the options to dotnet pack and dotnet run. However, they don't seem to work with dotnet test:
image

Did you really mean dotnet test above?

**Comment by KathleenDollard** [comment]
@livarcocc 

I'm a bit confused here. Why are these restore options supported, but not visible in --help? Feel free to grab me to discuss.

**Comment by mairaw** [comment]
@KathleenDollard on the issue describing this (#3388), @livarcocc said this was a bug and he was going to try to locate it.

**Comment by livarcocc** [comment]
We don't display the restore options in the commands that support implicit restore.

**Comment by mairaw** [comment]
@livarcocc @KathleenDollard do we still want to proceed with adding those options to the docs? Is this ok to merge or needs more work? Thanks,

**Comment by livarcocc** [comment]
I wonder if for these commands we should just say that they support the restore options when used with the long form. And maybe then, we can add some examples.

We explicitly didn't add it to the CLI help, because it would pollute the help output and draw from the options directly supported by the command being used. The way it is here, I fear it is mixing up restore and command options. Another option could be separating them a little in the document?

**Comment by mairaw** [comment]
We can go in any preferred direction @livarcocc. Maybe better to just add a note about the implicit restore and the options accepted and a few examples then?

/cc @KathleenDollard

**Comment by KathleenDollard** [comment]
I like the note about implicit restore, that the long form exists and a few examples.

**Comment by mairaw** [comment]
Looks good now @KathleenDollard @livarcocc? Test URLs are in the PR description.

**Comment by KathleenDollard** [comment]
Looks good to me

**Pull request merged**